### PR TITLE
[Week 1][How to Guide] Updated readme for How to Guide

### LINKED ISSUE #896: Update and enhance how-to-guide section
This issue is to track :
* What has been done and 
* What we are on currently working :)

## Team members
Abhishek Jaiswal - @abhishek-iiit 
Vasvi Sood  - @vasvi-sood 

# References 
- [x] **Readme(to github section)** 
- [x] **Overview** 
- [x] **Add events to the bot**
    - [x] Add events and customise a bot
- [x] **Add more functionality to your bot**
    - [x] Dealing with messages
    - [x] Manage Contacts
    - [x] Manage Rooms
    - [x] Send and Accept Friend Requests
    - [x] Sending and Receiving files
- [x] **Deploy on IM platfrom**
    - [x] Deploy in Wechat
    - [x] Deploy in Whatsapp
    - [x] Deploy in Lark
    - [x] Deploy in Gitter
    - [x] Deploy in Wechat Official Account
    - [x] Deploy in PadLocal
    - [x] Deploy in WeChat4U
- [x] **Overview with containers**
    - [x] Deploy with Heroku
    - [x] Deploy with Docker

Please link pull request to this section for the easy access.

### PR DESCRIPTION
**Pull request by vasvi-sood**
### Week 1
- [x] Get familiar with the project
- [x] Finalized and submitted the team project proposal 
- [x] Removed the About bot section and added Deploy with Heroku & Deploy with Docker in How-to Guide
- [x] Add Readme file to GitHub to help future contributors understand how to contribute and use How-to-guide section

Screenshot from 2021-05-26 18-11-35
Screenshot from 2021-05-26 18-12-20
Screenshot from 2021-05-26 18-12-33
Screenshot from 2021-05-26 18-12-44
Screenshot from 2021-05-26 18-12-59
Screenshot from 2021-05-26 18-13-09

### CODE REVIEW

**Review by nibble0101 — APPROVED** [review]
Thanks for adding this @vasvi-sood. This is very useful information.

**Comment by proudofsimin** [comment]
Hi Vasvi, thank you for your effort! Can you change the bulleted points into numbers, since they are sequential lists. Thank you.

**Comment by Rohitesh-Kumar-Jain** [comment]
> Screenshot from 2021-05-22 19-50-48


Thanks for adding screenshots in the description, it's good to add individual screenshots when you are modifying some part of the documentation, but if you are writing the documentation from scratch or modifying > 75% of the documentation I will prefer one single full-page screenshot. 

Another thing I feel is, we probably don't need too many images in the readme. I feel that images associated with git commit, and push are redundant, they probably won't help the reader much. What do you suggest @proudofsimin ? I will go with her views, as she is the one spearheading documentation style and structure : )

**Comment by vasvi-sood** [comment]
> Hi Vasvi, thank you for your effort! Can you change the bulleted points into numbers, since they are sequential lists. Thank you.

@proudofsimin  Done

**Comment by proudofsimin** [comment]
> ### Week 1
> * [x]  Get familiar with the project
> * [x]  Finalized and submitted the team project proposal
> * [x]  Removed the About bot section and added Deploy with Heroku & Deploy with Docker in How-to Guide
> * [x]  Add a README file to GitHub to help future contributors understand how to contribute to the How-to-guide page
> 
> Screenshot from 2021-05-22 19-50-21
> Screenshot from 2021-05-22 19-50-27
> Screenshot from 2021-05-22 19-50-34
> Screenshot from 2021-05-22 19-50-40
> Screenshot from 2021-05-22 19-50-48

**Review by huan — CHANGES_REQUESTED** [review]
Thank you for the great improvement!

I have the following review comments:

1. The index file of the folder should be **overview.mdx** instead of ~~README.md~~ (just like all other folders under the docs/)
2. All images in the docs page should be put under a folder with the same name of the docs page. For example:
    -  If we have a docs page at `docs/howto/overview.mdx`
    - Then the image should be put under the folder `static/img/docs/howto/overview/`
3. Please keep all file names to be lowercase

File: README.md [hunk 1]
```diff
@@ -120,13 +120,24 @@ At the same time, we also meet up offline all over the world. Here are some acti
...
 
 - Fork this repository
 - Create your documentation branch: `git checkout -b branch-name`
 - Make changes to the documentation in markdown
-- Commit your changes using the command `git commit -am 'commit message'`
-- Push your changes to gitHub
+- Add changes to the staging area `git add .`
+- Commit your changes using the command `git commit -m 'commit message'`
+- Push your changes to gitHub `git push origin branch-name`
 - Create new Pull Request
```
Comment (proudofsimin) [inline README.md]: Please change this part to numbered list as well. Thank you :)

**Review by proudofsimin — CHANGES_REQUESTED** [review]
It's great you changed the first part, but the second part is also a sequential list. Please change it as well. Thank you.

File: README.md (hunk 1 above)
Comment (Rohitesh-Kumar-Jain) [reply README.md]: Yeah numbered list will be better

**Comment by abhishek-iiit** [comment]
Thank you @huan for your feedback,

For clear understanding, We do have both the part for our how-to guide section
1. For the index file, we do have the overview.mdx file that can be viewed on the website where we will have the short link to all the sections under how-to-guide.
2. For the readme.md that can be viewed from GitHub, we were planned to add how to contribute and run it locally to this file.

and we will surely be taking care of the format and the files that are to be kept under different sections.
Please let us know more changes if needed, so we can improve our docs :)

**Comment by abhishek-iiit** [comment]
I request @huan, @Rohitesh-Kumar-Jain & @proudofsimin to please have a look at the changes. and give us the feedback for the same :)

**Comment by vasvi-sood** [comment]
@proudofsimin  @Rohitesh-Kumar-Jain all changes have been made. Kindly review the pull request and merge it.

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
Looks good for now, we will have a final review towards the end.
For now, I believe the PR is ready to get merged, well done.

**Review by proudofsimin — APPROVED** [review]

**Comment by Rohitesh-Kumar-Jain** [comment]
this PR is the root cause of all the evil 🤦🏼 

I realized now that this PR is being pushed directly from the master branch of vavsi-sood. (This doesn't cause a problem if you are working on a solo project).

This is the PR from all the refactoring should start : )

Surprisingly I ignored it the first time I reviewed this, and @proudofsimin and @nibble0101 also approved this 😅 

I am not a git expert but that's what I believe is happening here.

So in short you need to close all the 4 PRs you have raised in the GSoD so far. 

You don't need to close them yourself actually, the below manoeuvre will automatically close PRs and reset everything.

## fix your current master branch

First of all, check out to your master branch, and then checkout to a new PR branch based on that.

```
git switch master
git switch -c new-branch
```

Then push the new branch to your fork and create a new PR based on that branch (the PR will be identical to this one with the exception of the branch). When you've done that, we can reset your master branch.

```
git switch master
git fetch upstream master # fetch upstream/master without merging into local master
git reset --hard upstream/master # reset local master to upstream/master's state
git push --force origin master # force-update your fork's master to reflect upstream
```

I hope I am doing right @huan and not being harsh on the How-to-guides team ?

**Comment by Rohitesh-Kumar-Jain** [comment]
@abhishek-iiit start from this PR, first make sure that your master branch is even with the upstream master. 

Remember this rule: `Your local master should be never ahead of upstream master` your other branches can be ahead of upstream master.

**Comment by vasvi-sood** [comment]
We will surely make the necesaary changes. And submit the work again in the next 2-3 days.

**Pull request closed without merging**